libcomposefs/lcfs-writer-erofs.c

```diff
@@ -992,9 +992,10 @@ static int write_erofs_inode_data(struct lcfs_ctx_s *ctx, struct lcfs_node_s *no
 		for (size_t i = 0; i < node->n_xattrs; i++) {
 			struct lcfs_xattr_s *xattr = &node->xattrs[i];
 			if (xattr->erofs_shared_xattr_offset >= 0) {
-				uint64_t offset =
-					ctx_erofs->inodes_end % EROFS_BLKSIZ +
-					xattr->erofs_shared_xattr_offset;
+				uint64_t offset = ctx_erofs->inodes_end;
+				if (ctx->options->version < 2)
+					offset %= EROFS_BLKSIZ;
+				offset += xattr->erofs_shared_xattr_offset;
```

Comment:
I'm not particularly fond of how this conditional is done. It makes it quite complicated to understand what is going on. I think a more plain if (version < 2) { do it in old way} else { do it in new way } would be easier to understand.